torch/distributed/distributed_c10d.py

```diff
@@ -1284,11 +1284,16 @@ def _new_process_group_helper(
                 assert isinstance(
                     pg_options, ProcessGroupNCCL.Options
                 ), "Expected pg_options argument to be of type ProcessGroupNCCL.Options"
+                if pg_options._timeout:
+                    raise ValueError(
```

Comment:
do you want to just throw a waring instead of Error?

Reply:
yea, good point. If this discrepancy hasn't been bothering people, then they can take their time to fix their usage.  No point introducing a bunch of breakages for people.

Reply:
I dont think many people are using timeout in pg_options, so warning sounds good. If we really wanted to we could just save the timeout and then set it again after if it has a value.